src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -1524,30 +1512,72 @@ public long getMaxPayload() {
     }
 
     public Collection<String> getServers() {
-        NatsServerInfo info = this.serverInfo.get();
-        HashSet<String> check = new HashSet<>();
-        ArrayList<String> servers = new ArrayList<>();
-
-        options.getServers().stream().forEach(x -> {
-            String uri = x.toString();
-            if (!check.contains(uri)) {
-                servers.add(uri);
-                check.add(uri);
-            }
-        });
+        return getServersList();
+    }
+

```

Comment:
refactored so I could re-use the code. This method returns a generic collection, but was made from a list. At the same time it wanted an actual set so there would not be dupes, I already mentioned that I normalized so dupes would actually be found. But I think it was a list because ordering is maintained in a list, versus a set, where the order is not guaranteed. So I left it as a list.